libs/vscode/nx-project-view/src/lib/nx-project-tree-view.ts

```diff
@@ -0,0 +1,177 @@
+import { ProjectConfiguration } from '@nrwl/devkit';
+import { CliTaskProvider } from '@nx-console/vscode/tasks';
+import { getOutputChannel } from '@nx-console/vscode/utils';
+import { TreeItemCollapsibleState } from 'vscode';
+import { AbstractView, isDefined, TreeViewStrategy } from './nx-project-helper';
+import {
+  NxFolderTreeItem,
+  NxProjectTreeItem,
+  NxTreeViewItem,
+} from './nx-project-tree-item';
+
+type TreeViewMap = Map<string, [string, ProjectConfiguration][]>;
+
+export function createTreeViewStrategy(
+  cliTaskProvider: CliTaskProvider
+): TreeViewStrategy {
+  const listView = new TreeView(cliTaskProvider);
+  return {
+    getChildren: listView.getChildren.bind(listView),
+    getParent: listView.getParent.bind(listView),
+  };
+}
+
+class TreeView extends AbstractView {
+  constructor(cliTaskProvider: CliTaskProvider) {
+    super(cliTaskProvider);
+  }
+  async getParent(element: NxTreeViewItem) {
+    if (element instanceof NxFolderTreeItem) {
+      if (this.isRootPath(element.path)) {
+        return null;
+      }
+      const projectDefs = await this.cliTaskProvider.getProjects();
+      const map = this.groupByRootPath(projectDefs);
+      const parentFolders = this.getParentFolder(map, element.path);
+      return parentFolders.map(([path]) => this.createFolderTreeItem(path));
+    }
+
+    if (element instanceof NxProjectTreeItem) {
+      const projectRoot = element.nxProject.root;
+      const projectDefs = await this.cliTaskProvider.getProjects();
+      const map = this.groupByRootPath(projectDefs);
+      const parentFolders = this.getParentFolder(map, projectRoot);
+      return parentFolders.map(([path]) => this.createFolderTreeItem(path));
+    }
+
+    return this.getParentOfTargetItem(element);
+  }
+
+  async getChildren(element?: NxTreeViewItem) {
+    if (!element) {
+      return this.createFolders();
+    }
+    if (element instanceof NxFolderTreeItem) {
+      return this.createFolders(element);
+    }
+    if (element instanceof NxProjectTreeItem) {
+      return this.createTargetsFormProject(element);
+    }
+    return this.createConfigurationsFormTarget(element);
+  }
+
+  private async createFolders(parent?: NxFolderTreeItem) {
+    const projectDefs = await this.cliTaskProvider.getProjects();
+    const map = this.groupByRootPath(projectDefs);
+
+    if (!parent) {
+      const rootFolders = this.getRootFolders(map);
+      return rootFolders.map(([path]) => this.createFolderTreeItem(path));
+    }
+
+    const subFolders = this.getSubFolders(map, parent.path);
+    return subFolders.map(([path, projects]) =>
+      this.createFolderTreeItem(path, projects)
+    );
+  }
+
+  private createFolderTreeItem(
+    path: string,
+    projects?: [string, ProjectConfiguration][]
+  ) {
+    if (projects && projects.length === 1) {
+      const [project] = projects;
+      return this.createProjectTreeItem(project);
+    }
+
+    const folderName = path.split('/').pop() ?? '';
```

Comment:
Have you tested this in different OS? I feel like using a hardcoded `/` is prone to causing errors. I would rather use `path` utility methods to abstract things like splitting paths for us.

Reply:
Fixed, I used `path.sep` to get the platform specific separator and use it as split value.
https://nodejs.org/api/path.html#path_path_sep

For joning, I switched to `path.join`.
https://nodejs.org/api/path.html#pathjoinpaths

Reply:
I was just trying to validate it for windows and it does not yet work...
![image](https://user-images.githubusercontent.com/19486965/197607591-4fd33a2a-44c2-47e5-a4b1-2022ba27bbad.png)
I will look into it

Reply:
I noticed 2 things:
1. `TreeView.groupByRootPath` does not work under windows
2. `cliTaskProvider.getProjects()` returns different results depending on if the `node_modules` are installed or not.

The first one I will fix in a day or two.

The latter one should generally be fixed in the project, but I will only provide a workaround in this PR (to keep the PR clean).
It should look like:
```ts
// cli-task-provider.ts

export type NodeModulesNotInstalledWorkspaceProjects = {
  [project: string]: string;
};

// ...
async getProjects(
    json?: WorkspaceJsonConfiguration
  ): Promise<WorkspaceProjects | NodeModulesNotInstalledWorkspaceProjects> {
    if (json) {
      return json.projects;
    } else {
      const result = await nxWorkspace(getWorkspacePath(), outputLogger);
      if (!result.validWorkspaceJson || !result.workspace) {
        return {};
      } else {
        return result.workspace.projects;
      }
    }
  }
```

Reply:
Ok, then I will keep my consideration for this minimal. But I have to consider it at least bit, otherwise the tree view will show nothing at all in case that the node_modules are not installed.

If that is fine for you, then I will leave it as is. Otherwise, as said, I would try to keep it as minimal as possible

Reply:
I implemented the minimum, as mentioned

Reply:
Yes, it's fine if it shows nothing for now, thank you! 🙂

Reply:
I reverted the change, as it is fine to have it empty for now.
This also reduces the complexity of the PR 😁